iset.mm

```diff
@@ -81880,6 +81880,53 @@ Ordering on reals (cont.)
   0le1 $p |- 0 <_ 1 $=
     ( cc0 c1 0re 1re 0lt1 ltleii ) ABCDEF $.
 
+  ${
+    $d x B $.  $d x y C $.  $d x y D $.  $d x y M $.  $d x y N $.  $d x y ph $.
+    $d x y S $.
+    ltord.1 $e |- ( x = y -> A = B ) $.
+    ltord.2 $e |- ( x = C -> A = M ) $.
+    ltord.3 $e |- ( x = D -> A = N ) $.
+    ltord.4 $e |- S C_ RR $.
+    ltord.5 $e |- ( ( ph /\ x e. S ) -> A e. RR ) $.
+    ${
+      ltord.6 $e |- ( ( ph /\ ( x e. S /\ y e. S ) ) -> ( x < y -> A < B ) ) $.
+      $( Lemma for ~ eqord1 .  (Contributed by Mario Carneiro, 14-Jun-2014.) $)
+      ltordlem $p |- ( ( ph /\ ( C e. S /\ D e. S ) ) ->
+                        ( C < D -> M < N ) ) $=
+        ( clt wbr wi wceq cv wral wcel wa ralrimivva breq1 breq1d imbi12d breq2
+        eqeq1 eqeq1d chvarv breq2d rspc2v mpan9 ) ABUAZCUAZQRZDEQRZSZCHUBBHUBFH
+        UCGHUCUDFGQRZIJQRZSZAUTBCHHPUEUTVCFUQQRZIEQRZSBCFGHHUPFTZURVDUSVEUPFUQQ
+        UFVFDIEQLUGUHUQGTZVDVAVEVBUQGFQUIVGEJIQUPGTZDJTZSVGEJTZSBCUPUQTZVHVGVIV
+        JUPUQGUJVKDEJKUKUHMULUMUHUNUO $.
+
+      $( Infer an ordering relation from a proof in only one direction.
+         (Contributed by Mario Carneiro, 14-Jun-2014.)  (Revised by Jim
+         Kingdon, 20-Dec-2022.) $)
+      eqord1 $p |- ( ( ph /\ ( C e. S /\ D e. S ) ) -> ( C = D <-> M = N ) ) $=
```

Comment:
I don't understand the comment. As I understand the statement, you have a real function on S, which is $x \mapsto A(x)$, which is strictly increasing (by hypotheses 1 and 6). Its value at C (resp. D) is M (resp. N) by hypotheses 1 and 2 (resp. 1 and 3). So the theorem is:
> a real strictly increasing function on a subset of $\R$ is injective

(?)

Reply:
I think that's a better way to state it, but perhaps @digama0 who wrote the comment would see something I'm not seeing (either about whether that is accurate or about whether it emphasizes the wrong thing).

I'm not sure I'd design any of this exactly how `eqord1` does it (with so many hypotheses and classes), but then again I'm not sure what my alternative would be and anyway I'm usually in the business of intuitionizing things like this rather than revamping them.

Reply:
Looks like this (and the similar next) are the only two remaining points. Since you and I agree, if there is no objection from @digama0, I think you can adopt the wording I proposed.

Reply:
I've pushed up a change to use what is basically BJ's wording, for both set.mm and iset.mm. If we want to revise further we can always do so later.